sdk/storage/blobs/blob_client.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace azure { namespace storage { namespace blobs {
+
+class blob_client
+{
+public:
+    // connection string
+    blob_client(const std::string& connection_string, std::string container_name, std::string blob_name, blob_client_options options);
+
+    // shared key auth 
+    blob_client(azure::core::http::uri uri, azure::storage::common::storage_shared_key_credential credential, blob_client_options options);
+
+    // token auth
+    blob_client(azure::core::http::uri uri, azure::core::token_credential credential, blob_client_options options);
+
+    // anonymous/SAS/customized pipeline auth
+    blob_client(azure::core::http::uri uri, blob_client_options options);
+
+    bool is_snapshot() const;
```

Comment:
Why this is needed?

Reply:
Otherwise, how do we donate a blob snapshot?